flyteadmin/pkg/workflowengine/impl/k8s_executor.go

```diff
@@ -94,13 +93,27 @@ func (e K8sWorkflowExecutor) Abort(ctx context.Context, data interfaces.AbortDat
 	if err != nil {
 		return errors.NewFlyteAdminErrorf(codes.Internal, err.Error())
 	}
-	err = target.FlyteClient.FlyteworkflowV1alpha1().FlyteWorkflows(data.Namespace).Delete(ctx, data.ExecutionID.GetName(), v1.DeleteOptions{
-		PropagationPolicy: &deletePropagationBackground,
-	})
+
+	w, err := target.FlyteClient.FlyteworkflowV1alpha1().FlyteWorkflows(data.Namespace).Get(ctx, data.ExecutionID.GetName(), v1.GetOptions{})
+	if err != nil && !k8_api_err.IsNotFound(err) {
+		return errors.NewFlyteAdminErrorf(codes.Internal, "failed to terminate execution: %v with err %v", data.ExecutionID, err)
+	} else if k8_api_err.IsNotFound(err) {
+		logger.Infof(ctx, "Trying to abort an execution [%+v] that is not found in cluster: %s, skipping...", data.ExecutionID, target.ID)
+		return nil
+	}
+
+	if w.ObjectMeta.Annotations == nil {
+		w.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	w.ObjectMeta.Annotations[AbortedWorkflowAnnotation] = "true"
```

Comment:
concern here is if there's a case that this delete is required to delete dangling CRDs (not picked up by propeller)

Reply:
I guess.. that the dangling CRDs should be removed by propeller GC, right?

Reply:
It is the same case as completed workflows/failed workflows.

Reply:
>concern here is if there's a case that this delete is required to delete dangling CRDs (not picked up by propeller) 

If the concern is about deleting CRDs if the propeller is down.. what do you think about having a flag in the terminate to force the delete from the admin? Similar to kubectl delete --force which force the deletion of a CRD.